### PR TITLE
完成tutorial 和部分API reference 到2.4 版本文档的升级

### PR DESCRIPTION
**Pull request by chenghuzi**

### CODE REVIEW

File: source/locale/zh_CN/LC_MESSAGES/faq.po [hunk 1]
```diff
@@ -66,58 +66,56 @@ msgstr ""
...
 msgid ""
 "There are two ways of persisting studies, which depends if you are using "
 "in-memory storage (default) or remote databases (RDB). In-memory studies "
 "can be saved and loaded like usual Python objects using ``pickle`` or "
 "``joblib``. For example, using ``joblib``:"
 msgstr ""
 "有两种方法可以将 study 持久化。具体采用哪种取决于你是使用内存存储 (in-memory) 还是远程数据库存储 (RDB).通过 "
-"``pickle`` 或者 ``joblib``, 采用了内存存储的 study 可以和普通的 Python 对象一样，用 ``pickle`` 或者 ``joblib`` 来存储和加载。比如用 "
-"``joblib`` 的话："
+"``pickle`` 或者 ``joblib``, 采用了内存存储的 study 可以和普通的 Python 对象一样，用 ``pickle`` "
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/faq.po]: I think the `通过 ``pickle`` 或者 ``joblib``, ` part is redundant

File: source/locale/zh_CN/LC_MESSAGES/faq.po [hunk 2]
```diff
@@ -394,14 +394,20 @@ msgid ""
 " inside the objective function, except for the fact that "
 ":func:`gc.collect` is called even when errors, including "
 ":class:`~optuna.exceptions.TrialPruned` are raised."
-msgstr ""
+msgstr "运行垃圾回收器是有性能损失的，取决于你的目标函数运行的速度，这种损失可能是不能忽略的。"
+"因此，``gc_after_trial`` 在默认情况下是 :obj:`False`. 注意，上面这个例子类似于在目标函数内运行"
+"垃圾回收器，除了 :func:`gc.collect` 在出现包括 :class:`~optuna.exceptions.TrialPruned` "
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/faq.po]: I think `除了 :func:`gc.collect` 在出现包括 :class:`~optuna.exceptions.TrialPruned` 这样的错误时也会被调用以外。` maybe a little ambiguous.
How about `不同之处在于 :func:gc.collect 即使在出现错误甚至抛出 :class:~optuna.exceptions.TrialPruned时也会被调用`

File: source/locale/zh_CN/LC_MESSAGES/index.po [hunk 3]
```diff
@@ -81,8 +86,9 @@ msgid ""
 msgstr "采用了最先进的超参数采样和最有效的对无望 trial 进行剪枝的算法。"
 
 #: ../../source/index.rst:29
-msgid ":doc:`Easy parallelization <tutorial/004_distributed>`"
-msgstr ":doc:`并行的分布式优化 <tutorial/004_distributed>`"
+
+msgid ":doc:`Easy parallelization <tutorial/10_key_features/004_distributed>`"
+msgstr ":doc:`并行的分布式优化 <tutorial/10_key_features/004_distributed>`"
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/index.po]: 易用的并行优化？

File: source/reference/multi_objective/index.rst [hunk 4]
```diff
@@ -3,6 +3,8 @@
 optuna.multi_objective
 ======================
 
+This module is deprecated, with former functionality moved to :mod:`optuna.samplers`, :mod:`optuna.study`, :mod:`optuna.trial` and :mod:`optuna.visualization`.
```
Comment (belldandyxtq) [inline source/reference/multi_objective/index.rst:6]: Translation on this?

File: source/locale/zh_CN/LC_MESSAGES/tutorial/index.po [hunk 5]
```diff
@@ -23,97 +23,140 @@ msgstr "教程"
...
+#~ msgstr ""
+
+#~ msgid "**Other Resources**:"
+#~ msgstr ""
+
+#~ msgid ""
+#~ "`Examples`_: More examples including how "
+#~ "to use Optuna with popular libraries "
+#~ "for machine learning and deep learning."
+#~ msgstr ""
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/tutorial/index.po:137]: Translation?

File: source/locale/zh_CN/LC_MESSAGES/tutorial/index.po [hunk 6]
```diff
@@ -23,97 +23,140 @@ msgstr "教程"
...
-#: ../../source/tutorial/index.rst:227
+#: ../../source/tutorial/index.rst:268
 msgid "`Gallery generated by Sphinx-Gallery <https://sphinx-gallery.github.io>`_"
 msgstr ""
 
+#~ msgid ""
+#~ "Below tutorials cover the basic concepts"
+#~ " and usage of Optuna. The order "
+#~ "we assume is as follows:"
+#~ msgstr ""
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/tutorial/index.po:104]: ditto

File: source/locale/zh_CN/LC_MESSAGES/tutorial/index.po [hunk 7]
```diff
@@ -23,97 +23,140 @@ msgstr "教程"
...
-msgid ":ref:`cli`"
-msgstr ""
+"If you are new to Optuna or want a general introduction, we highly "
+"recommend the below video."
+msgstr "如果你不熟悉Optuna或者需要一个一般的介绍，我们强烈推荐下面这个视频。"
 
-#: ../../source/tutorial/index.rst:18
-msgid ":ref:`attributes`"
+#: ../../source/tutorial/index.rst:31
+msgid "Key Features"
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/tutorial/index.po:31]: ditto

File: source/locale/zh_CN/LC_MESSAGES/tutorial/index.po [hunk 8]
```diff
@@ -23,97 +23,140 @@ msgstr "教程"
...
 msgstr ""
 
-#: ../../source/tutorial/index.rst:19
-msgid ":ref:`pruning`"
+#: ../../source/tutorial/index.rst:33
+msgid ""
+"Showcases Optuna's `Key Features "
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/tutorial/index.po:36]: ditto

File: source/locale/zh_CN/LC_MESSAGES/tutorial/20_recipes/README.po [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+"Content-Transfer-Encoding: 8bit\n"
+"Generated-By: Babel 2.8.0\n"
+
+#: ../../source/tutorial/20_recipes/README.rst:4
+msgid "Recipes"
+msgstr ""
+
+#: ../../source/tutorial/20_recipes/README.rst:6
+msgid "Showcases the recipes that might help you using Optuna with comfort."
+msgstr ""
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/tutorial/20_recipes/README.po:26]: ditto

File: source/locale/zh_CN/LC_MESSAGES/tutorial/20_recipes/README.po [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+"Last-Translator: FULL NAME <EMAIL@ADDRESS>\n"
+"Language-Team: LANGUAGE <LL@li.org>\n"
+"MIME-Version: 1.0\n"
+"Content-Type: text/plain; charset=utf-8\n"
+"Content-Transfer-Encoding: 8bit\n"
+"Generated-By: Babel 2.8.0\n"
+
+#: ../../source/tutorial/20_recipes/README.rst:4
+msgid "Recipes"
+msgstr ""
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/tutorial/20_recipes/README.po]: ditto

File: source/locale/zh_CN/LC_MESSAGES/tutorial/20_recipes/sg_execution_times.po [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+"Last-Translator: FULL NAME <EMAIL@ADDRESS>\n"
+"Language-Team: LANGUAGE <LL@li.org>\n"
+"MIME-Version: 1.0\n"
+"Content-Type: text/plain; charset=utf-8\n"
+"Content-Transfer-Encoding: 8bit\n"
+"Generated-By: Babel 2.8.0\n"
+
+#: ../../source/tutorial/20_recipes/sg_execution_times.rst:7
+msgid "Computation times"
+msgstr ""
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/tutorial/20_recipes/sg_execution_times.po]: ditto

File: source/locale/zh_CN/LC_MESSAGES/reference/integration.po [hunk 12]
```diff
@@ -101,274 +106,338 @@ msgid "AllenNLP callback to prune unpromising trials."
...
+msgid ""
+":obj:`optuna.integration.BoTorchSampler "
+"<optuna.integration.BoTorchSampler>`"
+msgstr ""
+
+#: ../../source/reference/integration.rst:38:<autosummary>:1
+msgid ""
+"A sampler that uses BoTorch, a Bayesian optimization library built on top"
+" of PyTorch."
+msgstr ""
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/integration.po]: ditto

File: source/locale/zh_CN/LC_MESSAGES/reference/integration.po [hunk 13]
```diff
@@ -101,274 +106,338 @@ msgid "AllenNLP callback to prune unpromising trials."
...
+
+#: ../../source/reference/integration.rst:38:<autosummary>:1
+msgid ""
+":obj:`optuna.integration.botorch.qei_candidates_func "
+"<optuna.integration.botorch.qei_candidates_func>`"
+msgstr ""
+
+#: ../../source/reference/integration.rst:38:<autosummary>:1
+msgid "Quasi MC-based batch Expected Improvement (qEI)."
+msgstr ""
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/integration.po:133]: ditto

File: source/locale/zh_CN/LC_MESSAGES/reference/multi_objective/index.po [hunk 14]
```diff
@@ -17,10 +17,17 @@ msgstr ""
...
+#: ../../source/reference/multi_objective/index.rst:4
 msgid "optuna.multi_objective"
 msgstr ""
 
+#: ../../source/reference/multi_objective/index.rst:6
+msgid ""
+"This module is deprecated, with former functionality moved to "
+":mod:`optuna.samplers`, :mod:`optuna.study`, :mod:`optuna.trial` and "
+":mod:`optuna.visualization`."
+msgstr ""
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/multi_objective/index.po]: ditto

File: source/locale/zh_CN/LC_MESSAGES/reference/visualization/matplotlib.po [hunk 15]
```diff
@@ -25,57 +25,89 @@ msgstr ""
...
+msgid ""
+":obj:`optuna.visualization.matplotlib.plot_contour "
+"<optuna.visualization.matplotlib.plot_contour>`"
+msgstr ""
+
+#: ../../source/reference/visualization/matplotlib.rst:20:<autosummary>:1
+msgid ""
+"Plot the parameter relationship as contour plot in a study with "
+"Matplotlib."
+msgstr ""
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/visualization/matplotlib.po]: ditto

File: source/locale/zh_CN/LC_MESSAGES/reference/visualization/matplotlib.po [hunk 16]
```diff
@@ -25,57 +25,89 @@ msgstr ""
...
-#: ../../source/reference/visualization/matplotlib.rst:17:<autosummary>:1
+#: ../../source/reference/visualization/matplotlib.rst:20:<autosummary>:1
+msgid ""
+":obj:`optuna.visualization.matplotlib.plot_param_importances "
+"<optuna.visualization.matplotlib.plot_param_importances>`"
+msgstr ""
+
+#: ../../source/reference/visualization/matplotlib.rst:20:<autosummary>:1
+msgid "Plot hyperparameter importances with Matplotlib."
+msgstr ""
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/visualization/matplotlib.po]: ditto

File: source/locale/zh_CN/LC_MESSAGES/reference/visualization/matplotlib.po [hunk 17]
```diff
@@ -25,57 +25,89 @@ msgstr ""
...
+
+#: ../../source/reference/visualization/matplotlib.rst:20:<autosummary>:1
+msgid ""
+":obj:`optuna.visualization.matplotlib.plot_slice "
+"<optuna.visualization.matplotlib.plot_slice>`"
+msgstr ""
+
+#: ../../source/reference/visualization/matplotlib.rst:20:<autosummary>:1
+msgid "Plot the parameter relationship as slice plot in a study with Matplotlib."
+msgstr ""
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/visualization/matplotlib.po]: ditto

File: source/locale/zh_CN/LC_MESSAGES/tutorial/10_key_features/001_first.po [hunk 18]
```diff
@@ -0,0 +1,211 @@
...
+msgstr "在 Optuna 中，待优化函数一般被命名为 `objective`."
+
+#: ../../source/tutorial/10_key_features/001_first.rst:62
+msgid ""
+"This function returns the value of :math:`(x - 2)^2`. Our goal is to find"
+" the value of ``x`` that minimizes the output of the ``objective`` "
+"function. This is the \"optimization.\" During the optimization, Optuna "
+"repeatedly calls and evaluates the objective function with different "
+"values of ``x``."
+msgstr "这个函数返回:math:`(x - 2)^2`的值。我们的目标是找到"
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/tutorial/10_key_features/001_first.po]: The `:math:(x - 2)^2` is broken on the webpage

File: source/locale/zh_CN/LC_MESSAGES/tutorial/10_key_features/001_first.po [hunk 19]
```diff
@@ -0,0 +1,211 @@
+# SOME DESCRIPTIVE TITLE.
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/tutorial/10_key_features/001_first.po:1]: Did we translate this tutorial before?

**Review by belldandyxtq — CHANGES_REQUESTED** [review]

File: source/locale/zh_CN/LC_MESSAGES/tutorial/index.po (hunk 5 above)
Comment (chenghuzi) [reply source/locale/zh_CN/LC_MESSAGES/tutorial/index.po:137]: This part has been commented out

**Comment by chenghuzi** [comment]
The missing part in the reference section will be released in another PR.

File: source/locale/zh_CN/LC_MESSAGES/reference/integration.po [hunk 20]
```diff
@@ -191,7 +192,7 @@ msgstr ""
 msgid ""
 "A wrapper of :class:`~optuna.study.Study` to incorporate Optuna with "
 "ChainerMN."
-msgstr ""
+msgstr "一个将 Optuna 并入 CHainerMN 的 :class:`~optuna.study.Study` wrapper."
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/integration.po]: ChainerMN

**Review by belldandyxtq — CHANGES_REQUESTED** [review]
Sorry for being late

**Review by belldandyxtq — APPROVED** [review]

**Comment by toshihikoyanase** [comment]
I found a broken link in FAQ. The directory structure of optuna is updated in https://github.com/optuna/optuna/pull/2302.

```diff
66c66
< Please also refer to `sklearn_addtitional_args.py <https://github.com/optuna/optuna/blob/master/examples/sklearn_additional_args.py>`_ example,
---
> Please also refer to `sklearn_addtitional_args.py <https://github.com/optuna/optuna/blob/master/examples/sklearn/sklearn_additional_args.py>`_ example,
```

File: source/locale/zh_CN/LC_MESSAGES/tutorial/10_key_features/001_first.po [hunk 21]
```diff
@@ -0,0 +1,211 @@
...
+"This function returns the value of :math:`(x - 2)^2`. Our goal is to find"
+" the value of ``x`` that minimizes the output of the ``objective`` "
+"function. This is the \"optimization.\" During the optimization, Optuna "
+"repeatedly calls and evaluates the objective function with different "
+"values of ``x``."
+msgstr "这个函数返回 :math:`(x - 2)^2` 的值。我们的目标是找到"
+"``x``的值，使``目标``的输出最小化"
+"功能。这就是 "优化" 在优化过程中，Optuna"
+"反复调用并评估目标函数，用不同的"
+"``x```的数值。"
```
Comment (toshihikoyanase) [inline source/locale/zh_CN/LC_MESSAGES/tutorial/10_key_features/001_first.po]: In my environment, some symbols are not rendered as expected:

image

Is this my environmental issue? Or is it reproducible in your environment?

File: source/locale/zh_CN/LC_MESSAGES/tutorial/10_key_features/003_efficient_optimization_algorithms.po [hunk 22]
```diff
@@ -0,0 +1,333 @@
...
+
+#: ../../source/tutorial/10_key_features/003_efficient_optimization_algorithms.rst:28
+msgid "Optuna provides the following sampling algorithms:"
+msgstr "Optuna 提供了下列采样算法："
+
+#: ../../source/tutorial/10_key_features/003_efficient_optimization_algorithms.rst:30
+msgid ""
+"Tree-structured Parzen Estimator algorithm implemented in "
+":class:`optuna.samplers.TPESampler`"
+msgstr ":class:`optuna.samplers.TPESampler` 实现的树形的 Parzen Estimator 算法"
```
Comment (toshihikoyanase) [inline source/locale/zh_CN/LC_MESSAGES/tutorial/10_key_features/003_efficient_optimization_algorithms.po]: `Tree-structured Parzen Estimator` is a proper noun, and we might not need to translate it.

File: source/locale/zh_CN/LC_MESSAGES/tutorial/10_key_features/003_efficient_optimization_algorithms.po [hunk 23]
```diff
@@ -0,0 +1,333 @@
...
+"basically it is outperformed by "
+":class:`optuna.pruners.SuccessiveHalvingPruner` and "
+":class:`optuna.pruners.HyperbandPruner` as in `this benchmark result "
+"<https://github.com/optuna/optuna/wiki/%5BUnder-Construction%5D-"
+"Benchmarks-with-Kurobako>`_."
+msgstr "在大多数例子中我们采用的 :class:`optuna.pruners.MedianPruner`, 尽管其性能基本上会被 "
+":class:`optuna.pruners.SuccessiveHalvingPruner` 和 "
+":class:`optuna.pruners.HyperbandPruner` 超过，就像在 `这个基准测试结果 "
+"<https://github.com/optuna/optuna/wiki/%5BUnder-Construction%5D-"
+"Benchmarks-with-Kurobako>`_ 中那样."
```
Comment (toshihikoyanase) [inline source/locale/zh_CN/LC_MESSAGES/tutorial/10_key_features/003_efficient_optimization_algorithms.po]: This is not the translation issue, but this link is broken. Could you update it?
```suggestion
"<https://github.com/optuna/optuna/wiki/"
"Benchmarks-with-Kurobako>`_ 中那样."
```

File: source/locale/zh_CN/LC_MESSAGES/tutorial/10_key_features/003_efficient_optimization_algorithms.po [hunk 24]
```diff
@@ -0,0 +1,333 @@
...
+
+#: ../../source/tutorial/10_key_features/003_efficient_optimization_algorithms.rst:236
+msgid ""
+"From the benchmark results which are available at `optuna/optuna - wiki "
+"\"Benchmarks with Kurobako\" <https://github.com/optuna/optuna/wiki"
+"/%5BUnder-Construction%5D-Benchmarks>`_, at least for not deep learning "
+"tasks, we would say that"
+msgstr "对于非深度学习来说，根据 `optuna/optuna - wiki "
+"\"Benchmarks with Kurobako\" <https://github.com/optuna/optuna/wiki"
+"/%5BUnder-Construction%5D-Benchmarks>` 里的基准测试结果，我们推荐"
```
Comment (toshihikoyanase) [inline source/locale/zh_CN/LC_MESSAGES/tutorial/10_key_features/003_efficient_optimization_algorithms.po]: ditto.
```suggestion
"\"Benchmarks with Kurobako\" <https://github.com/optuna/optuna/wiki"
"/Benchmarks-with-Kurobako>` 里的基准测试结果，我们推荐"
```

Comment (toshihikoyanase) [reply source/locale/zh_CN/LC_MESSAGES/tutorial/10_key_features/003_efficient_optimization_algorithms.po]: Also, the link is not rendered as expected:

image

File: source/locale/zh_CN/LC_MESSAGES/tutorial/10_key_features/004_distributed.po [hunk 25]
```diff
@@ -0,0 +1,134 @@
...
+#: ../../source/tutorial/10_key_features/004_distributed.rst:35
+msgid "Create a Study"
+msgstr "创建 study"
+
+#: ../../source/tutorial/10_key_features/004_distributed.rst:37
+msgid ""
+"You can create a study using ``optuna create-study`` command. "
+"Alternatively, in Python script you can use :func:`optuna.create_study`."
+msgstr "你可以通过 ``optuna create-study`` 来创建 study."
+"或者在 Python 脚本hzong通过 :func:`optuna.create_study` 也行。"
```
Comment (toshihikoyanase) [inline source/locale/zh_CN/LC_MESSAGES/tutorial/10_key_features/004_distributed.po]: Not sure, but do we need `hzong` here?
```suggestion
"或者在 Python 脚本通过 :func:`optuna.create_study` 也行。"
```

File: source/locale/zh_CN/LC_MESSAGES/tutorial/20_recipes/004_cli.po [hunk 26]
```diff
@@ -0,0 +1,149 @@
...
+
+#: ../../source/tutorial/20_recipes/004_cli.rst:29
+msgid "Optuna provides command-line interface as shown in the above table."
+msgstr "Optuna 提供的命令行界面包括上表中的所有命令。"
+
+#: ../../source/tutorial/20_recipes/004_cli.rst:31
+msgid ""
+"Let us assume you are not in IPython shell and writing Python script "
+"files instead. It is totally fine to write scripts like the following:"
+msgstr "如果你不使用 Ipython shell, 而是写 Python 脚本文件的话，编写像下面这样的脚本是完全可以的："
```
Comment (toshihikoyanase) [inline source/locale/zh_CN/LC_MESSAGES/tutorial/20_recipes/004_cli.po]: A nit.
```suggestion
msgstr "如果你不使用 IPython shell, 而是写 Python 脚本文件的话，编写像下面这样的脚本是完全可以的："
```

**Review by toshihikoyanase — APPROVED** [review]
Thank you for your PR! I added some small comments mainly for syntax and broken links.

**Review by toshihikoyanase — CHANGES_REQUESTED** [review]
Oops! I chose the wrong option. I'm sorry for bothering you.

**Comment by toshihikoyanase** [comment]
@chenghuzi I'm sorry, but I wrongly approved this PR. Could you address the suggestions in https://github.com/optuna/doc-optuna-zh-cn/pull/3#pullrequestreview-597396121, please?

**Comment by chenghuzi** [comment]
> @chenghuzi I'm sorry, but I wrongly approved this PR. Could you address the suggestions in [#3 (review)](https://github.com/optuna/doc-optuna-zh-cn/pull/3#pullrequestreview-597396121), please?

Sure, will fix in next commit

File: source/locale/zh_CN/LC_MESSAGES/tutorial/10_key_features/001_first.po (hunk 21 above)
Comment (chenghuzi) [reply source/locale/zh_CN/LC_MESSAGES/tutorial/10_key_features/001_first.po]: Fixed, due to the lack of backslash and spaces

File: source/locale/zh_CN/LC_MESSAGES/tutorial/10_key_features/003_efficient_optimization_algorithms.po (hunk 22 above)
Comment (chenghuzi) [reply source/locale/zh_CN/LC_MESSAGES/tutorial/10_key_features/003_efficient_optimization_algorithms.po]: Agreed

File: source/locale/zh_CN/LC_MESSAGES/tutorial/10_key_features/004_distributed.po (hunk 25 above)
Comment (chenghuzi) [reply source/locale/zh_CN/LC_MESSAGES/tutorial/10_key_features/004_distributed.po]: Sorry, missed that typo in Pinyin

File: source/locale/zh_CN/LC_MESSAGES/tutorial/20_recipes/004_cli.po (hunk 26 above)
Comment (chenghuzi) [reply source/locale/zh_CN/LC_MESSAGES/tutorial/20_recipes/004_cli.po]: Fixed

File: source/locale/zh_CN/LC_MESSAGES/tutorial/10_key_features/001_first.po (hunk 21 above)
Comment (toshihikoyanase) [reply source/locale/zh_CN/LC_MESSAGES/tutorial/10_key_features/001_first.po]: Thanks! The rendering issue was resolved.

image

File: requirements.txt [hunk 27]
```diff
@@ -10,4 +10,7 @@ sphinxcontrib-serializinghtml==1.1.4
 sphinx-gallery
 matplotlib
 pillow
-scikit-learn
\ No newline at end of file
+scikit-learn
+pandas
+plotly>=4.0.0
+lightgbm
```
Comment (toshihikoyanase) [inline requirements.txt:16]: The sphinx directive `plotly` seems to be missing. Could you add `sphinx-plotly-directive` to `requirements.txt` and `source/conf.py`?

It was introduced by https://github.com/optuna/optuna/pull/1944, and included in v2.4.0.

c.f.,
https://github.com/optuna/optuna/blob/bf60fab2473599188c2722fff1f743d372412331/docs/source/conf.py#L59
https://github.com/optuna/optuna/blob/bf60fab2473599188c2722fff1f743d372412331/setup.py#L74

The warning message related to the missing directive is as follows:

```
reading sources... [100%] tutorial/20_recipes/004_cli
 /usr/local/lib/python3.7/site-packages/optuna/multi_objective/visualization/_pareto_front.py:docstring of optuna.multi_objective.visualization.plot_pareto_front:7: WARNING: Unknown directive type "plotly". 

```

**Review by toshihikoyanase — CHANGES_REQUESTED** [review]
Thank you for your update. I found another issue during the build.

File: requirements.txt (hunk 27 above)
Comment (toshihikoyanase) [reply requirements.txt:16]: Let us work on this in a follow-up PR.

**Review by toshihikoyanase — APPROVED** [review]
LGTM. Thank you for accomplishing the long-running PR.

**Pull request merged**